src/training/calculatorreview/Main.java

```diff
@@ -4,9 +4,79 @@
 
 public class Main {
 
+    private static Scanner in;
+
     public static void main(String[] args) {
+        printWelcomeMessage();
+
+        in = new Scanner(System.in);
+
+        while (true) {
+            performOneCalculation();
+        }
+    }
+
+    private static void performOneCalculation() {
+        String op = askForOperator();
+        int[] n = askForNumberArray(op);
+        int a = calculateAnswer(op, n);
+
+        System.out.println("The answer is: " + a);
+        System.out.println();
+    }
+
+    private static void printWelcomeMessage() {
         System.out.println("Welcome to the calculator!");
         System.out.println("==========================");
-        new Scanner(System.in);
+    }
+
+    private static String askForOperator() {
+        String[] allowedOperators = { "+", "-", "*", "/" };
+        System.out.print("Please enter the operator: ");
+        String op = in.nextLine();
+        return op;
+    }
+
+    private static int[] askForNumberArray(String op) {
+        System.out.print("How many numbers do you want to " + op + "? ");
+        int count = Integer.parseInt(in.nextLine());
+        int numbers[] = new int[]{count};
+
+        for(int index = 0; index < count; index++) {
+            numbers[index] = addNumber("Please enter number " + (index + 1) + ": ");
+        }
+        return numbers;
+    }
+
+    private static int addNumber(String message) {

```

Comment:
This function is named in a way which could be confusing to people reading through the code. Maybe name this something along the lines of what it is doing: asking the user a question and parsing their input to int.